keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -1172,6 +1207,179 @@ back to aggregating container requests.
+* Container c2 (specified memory limit): c2 does define container-level resources,
+  so the effective memory limit of c2 is the container level limit. Therefore, a
+  resize of the pod-level memory limit doesn't change the effective container limit,
+  so the c2 is not restarted when the pod-level memory limit is resized.
+
+##### Implementation Details
+
+###### Allocating Pod-level Resources
+Allocation of pod-level resources will work the same as container-level resources. The allocated resources checkpoint will be extended to include pod-level resources, and the pod object will be updated with the allocated resources in the pod sync loop.
+
+###### Actuating Pod-level Resource Resize
+A dirty bit for pod-level resources will be added to the allocation checkpoint to
+signal the need for resize.  This change, along with the existing tracking and
+triggering mechanisms, is sufficient for pod-level resize actuation.  The same
+ordering rules for pod and container resource resizing will be applied for each
+resource as needed:
+1. Increase pod-level cgroup (if needed)
+2. Decrease container resources
+3. Decrease pod-level cgroup (if needed)
+4. Increase container resources
+
+###### Tracking Actual Pod-level Resources
+To accurately track actual pod-level resources during in-place pod resizing, several
+changes are required:
+
+1. Configuration reading: Pod-level resource config is currently read as part of the
+   resize flow, but will also need to be read during pod creation. Critically, the
+   configuration must be read again after the resize operation to capture the
+   updated resource values.  Currently, the configuration is only read before a
+   resize.
+   
+2. Pod Status Update: Because the pod status is updated before the resize takes
+   effect, the status will not immediately reflect the new resource values.  If a
+   container within the pod is also being resized, the container resize operation
+   will trigger a pod synchronization (pod-sync), which will refresh the pod's
+   status.  However, if only pod-level resources are being resized, a pod-sync must
+   be explicitly triggered to update the pod status with the new resource
+   allocation.
+
+3. [Scoped for Beta] Caching: Actual pod resource data may be cached to minimize API server load. This cache, if implemented, must be invalidated after each successful pod resize to ensure that subsequent reads retrieve the latest information. The need for and implementation of this caching mechanism will be evaluated in the beta phase.  Performance benchmarking will be conducted to determine if caching is required and, if so, what caching strategy is most appropriate.
+
+**Note for future enhancements for Ephemeral containers with pod-level resources and
+IPPR**
+Previously, assigning resources to ephemeral
+containers wasn't allowed because pod resource allocations were immutable. With
+the introduction of in-place pod resizing, users will gain more flexibility:
```

Comment:
nit: since we're talking about a future enhancement, not the current proposal
```suggestion
the introduction of in-place pod resizing, users could gain more flexibility:
```